clients/venice-thin-client/src/main/java/com/linkedin/venice/client/schema/RouterBackedSchemaReader.java

```diff
@@ -726,4 +736,20 @@ private Schema preemptiveSchemaVerification(Schema writerSchema, String writerSc
     }
     return alternativeWriterSchema;
   }
+
+  private void cacheValueAndCanonicalSchemas(Schema valueSchema, int valueSchemaId) {
+    String canonicalSchemaStr = AvroCompatibilityHelper.toParsingForm(valueSchema);
+    Schema canonicalSchema = AvroSchemaParseUtils.parseSchemaFromJSONLooseValidation(canonicalSchemaStr);
+
+    cacheValueAndCanonicalSchemas(valueSchema, canonicalSchema, valueSchemaId);
+  }
+
+  private void cacheValueAndCanonicalSchemas(Schema valueSchema, Schema canonicalSchema, int valueSchemaId) {
+    valueSchemaToCanonicalSchemaId.put(valueSchema, valueSchemaId);
+
+    Integer cachedCanonicalSchemaId = canonicalValueSchemaMapR.getIfPresent(canonicalSchema);
+    if (cachedCanonicalSchemaId == null || cachedCanonicalSchemaId < valueSchemaId) {
```

Comment:
What is this case that cached schema id < value schema id? I am a bit confused on this.

Reply:
I wanted to have a case where we prioritize newer schemas in case the canonical schemas are same for multiple value schemas

Reply:
OK got it. Thanks!
(Would be nice to have 1 line doc to describe this so others won't get confused about this check condition, as it is not very obvious from the context.)

Reply:
Yeah. Good idea. I just updated it